### PR TITLE
[Vision & Multimodal] Component Updates & Stable Diffusion DreamBooth Finetuning Component

### PR DESCRIPTION
**Pull request by rjaincc**
1/ Adding Stable Diffusion text to image finetuning Components. 
2/ Update environment for vulnerabilities
3/ Update component versions for using latest environments

### CODE REVIEW

**Review by shubhamiit — COMMENTED** [review]
Instance prompt is non optional, please remove it from double brackets in ft spec.yaml

**Comment by shubhamiit** [comment]
> Instance prompt is non optional, please remove it from double brackets in ft spec.yaml

or make it optional.

File: assets/training/finetune_acft_multimodal/components/pipeline_components/multimodal_classification/spec.yaml [hunk 1]
```diff
@@ -387,7 +387,7 @@ jobs:
 
   multimodal_classification_finetune:
     type: command
-    component: azureml:multimodal_classification_finetune:0.0.2
+    component: azureml:multimodal_classification_finetune@latest #:0.0.2
```
Comment (nvijayrania) [inline assets/training/finetune_acft_multimodal/components/pipeline_components/multimodal_classification/spec.yaml:390]: Would "latest" work? I thought we need to specify the exact version

File: assets/training/finetune_acft_multimodal/components/model_import/multimodal_classification/spec.yaml [hunk 2]
```diff
@@ -8,7 +8,8 @@ description: Import PyTorch / MLflow model
 
 is_deterministic: True
 
-environment: azureml://registries/azureml/environments/acft-multimodal-gpu/versions/15
+# environment: azureml://registries/azureml/environments/acft-multimodal-gpu/versions/15
+environment: azureml:acft-multimodal-gpu@latest
```
Comment (nvijayrania) [inline assets/training/finetune_acft_multimodal/components/model_import/multimodal_classification/spec.yaml:12]: Would latest work here?

File: assets/training/vision/environments/automl-dnn-vision-gpu/context/Dockerfile [hunk 3]
```diff
@@ -20,7 +20,7 @@ RUN conda env create -p $AZUREML_CONDA_ENVIRONMENT_PATH -f conda_dependencies.ya
     conda clean -a -y
 
 # vulnearbility fix
-RUN pip install pyarrow==14.0.1
+RUN pip install pyarrow==14.0.2
```
Comment (nvijayrania) [inline assets/training/vision/environments/automl-dnn-vision-gpu/context/Dockerfile:23]: nit: Wondering 14.0.2 specifically? In other envs, we have 14.0.1 though.

File: assets/training/finetune_acft_multimodal/components/pipeline_components/multimodal_classification/spec.yaml (hunk 1 above)
Comment (rjaincc) [reply assets/training/finetune_acft_multimodal/components/pipeline_components/multimodal_classification/spec.yaml:390]: Thanks for catching that. Fixed.

File: assets/training/vision/environments/automl-dnn-vision-gpu/context/Dockerfile (hunk 3 above)
Comment (rjaincc) [reply assets/training/vision/environments/automl-dnn-vision-gpu/context/Dockerfile:23]: From envs/Latest/AzureML-AutoML-DNN-Vision-GPU/context/Dockerfile

**Review by nvijayrania — APPROVED** [review]

**Review by gauravrajguru — APPROVED** [review]

**Pull request merged**